server/parsec/components/postgresql/utils.py

```diff
@@ -95,36 +131,46 @@ def _q(
         suffix: str | None = None,
         **kwargs: Any,
     ) -> str:
-        if table_alias:
-            from_table = f"{table} as {table_alias}"
-            select_table = table_alias
-        else:
-            from_table = table
-            select_table = table
+        # Use table alias if specified
+        from_table = f"{table} AS {table_alias}" if table_alias else table
+        select_table = table_alias if table_alias else table
+
+        # Use either internal ID (_id) or public ID for the WHERE condition
         if _id is not None:
             condition = f"{select_table}._id = {_id}"
         else:
-            public_id = kwargs.pop(public_id_field, None)
+            public_id = kwargs.pop(public_id_column, None)
             assert public_id is not None
+
+            # When using public ID, the organization should also be specified.
+            # If the organization's internal ID (`organization`) is not specified,
+            # the organization's public ID (`organization_id`) is used to retrieve it.
             if organization is None:
                 assert organization_id is not None
                 organization = q_organization_internal_id(organization_id)
-            else:
-                assert organization is not None
-            condition = f"{select_table}.organization = {organization} AND {select_table}.{ public_id_field } = { public_id }"
+
+            condition = " AND ".join(
+                (
+                    f"{select_table}.{public_id_column} = {public_id}",
+                    f"{select_table}.organization = {organization}",
+                )
+            )
```

Comment:
Minor comment but I find join for two elements confusing. Maybe use 
```suggestion
            filter1 = f"{select_table}.{public_id_column} = {public_id}"
            filter2 = f"{select_table}.organization = {organization}"
            condition = f"{filter1} AND {filter2}"
```